src/mgr/ActivePyModules.cc

```diff
@@ -1516,16 +1517,35 @@ void ActivePyModules::register_client(std::string_view name, std::string addrs)
   entity_addrvec_t addrv;
   addrv.parse(addrs.data());
 
-  dout(7) << "registering msgr client handle " << addrv << dendl;
-  py_module_registry.register_client(name, std::move(addrv));
+  dout(7) << "registering msgr client handle [" << name << "] " << addrv << dendl;
+  py_module_registry.register_client(name, addrv);
+  while (true) {
+    // wait to receive new MgrMap from the monitor
+    cluster_state.wait_for_mgrmap();
+    if (cluster_state.is_client_in_mgrmap(name, addrv)) {
```

Comment:
Here you need to check if it's in the map, and if it's not, you need the epoch of the mgrmap that was inspected so you can correctly wait again. Otherwise you have races with mgrmap updates and your check.

Reply:
I think I've done what you suggested here. I'm not resolving this comment as I'm not completely sure.